extensions/amp-subscriptions/0.1/dialog-renderer.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../src/services';
+import {evaluateExpr} from './expr';
+
+
+export class DialogRenderer {
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!./dialog.Dialog} dialog
+   */
+  constructor(ampdoc, dialog) {
+    /** @private @const {!../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampdoc_ = ampdoc;
+
+    /** @private @const {!./dialog.Dialog} */
+    this.dialog_ = dialog;
+
+    /** @private @const {!../../../src/service/template-impl.Templates} */
+    this.templates_ = Services.templatesFor(ampdoc.win);
+  }
+
+  /**
+   * @param {!JsonObject} authResponse
+   * @return {!Promise<boolean>}
+   */
+  render(authResponse) {
+    // Make sure the document is fully parsed.
+    return this.ampdoc_.whenReady().then(() => {
+      // Find the first matching dialog.
+      const candidates = this.ampdoc_.getRootNode()
+          .querySelectorAll('[subscriptions-dialog][subscriptions-display]');
+      for (let i = 0; i < candidates.length; i++) {
+        const candidate = candidates[i];
+        const expr = candidate.getAttribute('subscriptions-display');
+        if (expr && evaluateExpr(expr, authResponse)) {
```

Comment:
Do we expect `authResponse` to be something like `{value: X}`? if Yes is this the best name for this variable?
Do we have a defined structure of authResponse?

Reply:
Yes. It's essentially the `Entitlement.json()` call. I just didn't get there yet. The expressions themselves do not care - they simply built around json.